### PR TITLE
fix: #4514 augment interfaces the same way no matter if they are used as the target of a relationship or not

### LINKED ISSUE #4514: Inconsistent augmentation of relational fields of interfaces
**Describe the bug**

Currently interfaces get augmented differently depending if they are used  as target of a relations or not.

Example

https://github.com/neo4j/graphql/blob/7342feda553c018f869a0cef4a070409155766d2/packages/graphql/tests/schema/directives/settable.test.ts#L7126-L7147

generates

https://github.com/neo4j/graphql/blob/7342feda553c018f869a0cef4a070409155766d2/packages/graphql/tests/schema/directives/settable.test.ts#L7553-L7558

but 

https://github.com/neo4j/graphql/blob/387e455bb9aebb94663b05baed6c8694ec304e80/packages/graphql/tests/schema/experimental-schema/interfaces.test.ts#L28-L46

generates

https://github.com/neo4j/graphql/blob/387e455bb9aebb94663b05baed6c8694ec304e80/packages/graphql/tests/schema/experimental-schema/interfaces.test.ts#L520-L525

where the arguments on the augmented relational fields of the interface `MovieNode` are missing.

**Expected behavior**

All augmented interfaces have arguments on their relational fields

### PR DESCRIPTION
**Pull request by Andy2003**
# Description

augment interfaces the same way no matter if they are used as the target of a relationship or not

## Complexity

Complexity: Low

# Issue

Closes #4514

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [x] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

File: packages/graphql/tests/schema/interfaces.test.ts [hunk 1]
```diff
@@ -487,11 +487,11 @@ describe("Interfaces", () => {
               id_EQUAL: ID @deprecated(reason: \\"Aggregation filters that are not relying on an aggregating function will be deprecated.\\")
             }
 
-            interface MovieNode @something(something: \\"test\\") {
+            interface MovieNode {
```
Comment (a-alle) [inline packages/graphql/tests/schema/interfaces.test.ts]: the directive is missing here, user defined directives are not propagated anymore

File: packages/graphql/src/schema/make-augmented-schema.ts [hunk 2]
```diff
@@ -437,56 +434,18 @@ function makeAugmentedSchema({
...
-            const userDefinedInterfaceDirectives = userDefinedDirectivesForInterface.get(entity.name) || [];
-            const propagatedDirectives = propagatedDirectivesForNode.get(entity.name) || [];
             const interfaceEntityAdapter = new InterfaceEntityAdapter(entity);
-            withInterfaceType({
-                entityAdapter: interfaceEntityAdapter,
-                userDefinedFieldDirectives,
-                userDefinedInterfaceDirectives,
+            generateInterfaceObjectType({
                 composer,
-                config: {
```
Comment (a-alle) [inline packages/graphql/src/schema/make-augmented-schema.ts:454]: I think this argument is obsolete now and can be removed

File: packages/graphql/src/schema/make-augmented-schema.ts [hunk 3]
```diff
@@ -437,56 +434,18 @@ function makeAugmentedSchema({
             return;
         }
         if (entity instanceof InterfaceEntity && !seenInterfaces.has(entity.name)) {
-            const userDefinedFieldDirectives = userDefinedFieldDirectivesForNode.get(entity.name) as Map<
-                string,
-                DirectiveNode[]
-            >;
-            const userDefinedInterfaceDirectives = userDefinedDirectivesForInterface.get(entity.name) || [];
```
Comment (a-alle) [inline packages/graphql/src/schema/make-augmented-schema.ts:446]: these directives are not handled anymore (`MovieNode` case in the tests)

**Review by a-alle — CHANGES_REQUESTED** [review]
Contribution much appreciated! Let's get the user defined interface directives on the generated type and get this in.

**Comment by Andy2003** [comment]
The tests are currently failing b/c of a bug I fixed in https://github.com/neo4j/graphql/pull/4509, wich should be merged 1st

**Comment by Andy2003** [comment]
The tests are currently failing b/c of a bug I fixed in https://github.com/neo4j/graphql/pull/4509, wich should be merged 1st

**Review by a-alle — APPROVED** [review]

**Pull request merged**